packages/frontend/src/testing/__mocks__/providers/AppProvider.mock.tsx

```diff
@@ -0,0 +1,50 @@
+import { Ability } from '@casl/ability';
+import {
+    ApiError,
+    HealthState,
+    LightdashUserWithAbilityRules,
+} from '@lightdash/common';
+import { useQuery } from '@tanstack/react-query';
+import { FC, PropsWithChildren } from 'react';
+import { UserWithAbility } from '../../../hooks/user/useUser';
+import { AppProviederContext } from '../../../providers/AppProvider';
+import mockHealthResponse from '../api/healthResponse.mock';
+import { mockUserResponse } from '../api/userResponse.mock';
+
+export type AppProviderMockProps = {
+    mocks?: {
+        health?: Partial<HealthState>;
+        user?: Partial<LightdashUserWithAbilityRules>;
+    };
+};
+
+const AppProviderMock: FC<PropsWithChildren<AppProviderMockProps>> = ({
+    children,
+    mocks,
+}) => {
+    const health = useQuery<HealthState, ApiError>({
+        queryKey: ['health'],
+        queryFn: () => mockHealthResponse(mocks?.health),
+    });
+
+    const user = useQuery<UserWithAbility, ApiError>({
+        queryKey: ['user'],
+        queryFn: () => {
+            const userResponse = mockUserResponse(mocks?.user);
+
+            return {
+                ...userResponse,
+                ability: new Ability(userResponse.abilityRules),
+            };
+        },
+        enabled: !!health.data?.isAuthenticated,
+    });
+
+    return (
+        <AppProviederContext.Provider value={{ health, user }}>
```

Comment:
this file mocks real AppProvider by exposing mocked user and health data. it also accepts overrides as properties. that's how `renderWithProviders` and `renderHookWithProviders` produce mocked render.